### PR TITLE
Prevent the same variable to be evaluated multiple times in on_change function (#257)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #257 

With this udpate, variable x in the on_change function wont be updated in this case since x has been updated before --> No inifinite loop.

```
from taipy.gui import Gui

x = 10

def on_change(state, var, val):
    if var == "x":
        state.x = val * 2

gui = Gui(page="""

x = <|{x}|>

x * 10 = <|{x*10}|>

Slider for x: <|{x}|slider|>

""")

gui.run()
```

### CODE REVIEW

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -190,6 +190,9 @@ def __init__(
         self.on_change: t.Optional[t.Callable] = None
         self.on_init: t.Optional[t.Callable] = None
 
+        # gui syncrhonous state variable
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py]: typo in synchronous

**Review by FredLL-Avaiga — APPROVED** [review]
Well done

File: taipy/gui/gui.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: thanks

**Review by FabienLelaquais — APPROVED** [review]
Simple enough. Good job!

**Pull request merged**